src/zarr/abc/store.py

```diff
@@ -11,6 +11,7 @@
 
 
 class AccessMode(NamedTuple):
+    str: AccessModeLiteral
```

Comment:
Do we want to shadow the builtin `str`?

Reply:
I don't see an issue here. This simply allows us to extract the mode as a string from the named tuple like this: `mode.str`. I think this is a fairly common pattern.

Reply:
I didn't see that this was a member of a named tuple. seems legit